contracts/src/Rollup.sol

```diff
@@ -177,16 +177,17 @@ contract Rollup is RollupBase {
         if (staker.assertionID > lastConfirmedAssertionID) {
             revert StakedOnUnconfirmedAssertion();
         }
+        uint256 stakedAmount = staker.amountStaked;
         deleteStaker(stakerAddress);
         // Note: we don't need to modify assertion state because you can only unstake from a confirmed assertion.
-        (bool success,) = stakerAddress.call{value: staker.amountStaked}("");
+        (bool success,) = stakerAddress.call{value: stakedAmount}("");
```

Comment:
you can just swap the order of these two lines instead right

Reply:
Sure, done. Please review again.